tests/conftest.py

```diff
@@ -231,6 +231,62 @@ def _benchmark_time():
     return _benchmark_time()
 
 
+@pytest.fixture(scope="function")
+def benchmark_coiled_prometheus(test_run_benchmark):
+    """Benchmark the coiled prometheus metrics of executing some code.
+
+    Yields
+    ------
+    Context manager that records the coiled prometheus metrics of executing
+    the ``with`` statement if run as part of a benchmark, or does nothing otherwise.
+
+    Example
+    -------
+    .. code-block:: python
+
+        def test_something(benchmark_coiled_prometheus):
+            with benchmark_coiled_prometheus:
+                do_something()
+    """
+
+    @contextlib.contextmanager
+    def _benchmark_coiled_prometheus(client):
+        if not test_run_benchmark:
+            yield
+        else:
+            start = time.time()
+            yield
+            end = time.time()
+            cluster = client.cluster
+            test_run_benchmark.scheduler_memory_max = cluster.get_aggregated_metric(
+                query="(host_memory_total-host_memory_available)&scheduler",
+                over_time="max",
+                start_ts=start,
+                end_ts=end,
+            )
+            test_run_benchmark.scheduler_cpu_avg = cluster.get_aggregated_metric(
+                query="cpu&scheduler",
+                over_time="avg",
+                start_ts=start,
+                end_ts=end,
+            )
+            # TODO: Do we want to have plain max or 80th percentile?
```

Comment:
I'm not entirely sure what we're interested in, yet. My sense is that we do not want to be alarmed if there is a single peak (e.g. single gc.collect takes longer) but if a high quantile goes up we should be concerened

Reply:
95% quantile on some very short tests seems very aggressive to me; 80% sounds more reasonable IMHO